presto-main/src/main/java/io/prestosql/operator/DynamicFilterCollectOperator.java

```diff
@@ -0,0 +1,207 @@
+import io.prestosql.spi.Page;
+import io.prestosql.spi.block.Block;
+import io.prestosql.spi.predicate.Domain;
+import io.prestosql.spi.predicate.TupleDomain;
+import io.prestosql.spi.predicate.ValueSet;
+import io.prestosql.spi.type.Type;
+import io.prestosql.spi.type.TypeUtils;
+import io.prestosql.sql.DynamicFilters;
+import io.prestosql.sql.planner.LocalDynamicFilter;
+import io.prestosql.sql.planner.Symbol;
+import io.prestosql.sql.planner.TypeProvider;
+import io.prestosql.sql.planner.plan.PlanNodeId;
+import io.prestosql.sql.tree.SymbolReference;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.Verify.verify;
+import static io.prestosql.SystemSessionProperties.getDynamicFilteringMaxPerDriverByteSize;
+import static io.prestosql.SystemSessionProperties.getDynamicFilteringMaxPerDriverValueCount;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * This operator acts as a simple "pass-through" pipe, while saving its input pages.
+ * The collected pages' value are used for creating a run-time filtering constraint (for probe-side table scan in an inner join).
+ * We support only small build-side pages (which should be the case when using "broadcast" join).
+ */
+public class DynamicFilterCollectOperator
+        implements Operator
+{
+    public static class DynamicFilterCollectOperatorFactory
+            implements OperatorFactory
+    {
+        private final int operatorId;
+        private final PlanNodeId planNodeId;
+        private final LocalDynamicFilter dynamicFilter;
+        private final TypeProvider types;
+
+        public DynamicFilterCollectOperatorFactory(int operatorId, PlanNodeId planNodeId, LocalDynamicFilter dynamicFilter, TypeProvider types)
+        {
+            this.operatorId = operatorId;
+            this.planNodeId = requireNonNull(planNodeId, "planNodeId is null");
+            this.dynamicFilter = requireNonNull(dynamicFilter, "dynamicFilter is null");
+            this.types = requireNonNull(types, "types is null");
+        }
+
+        @Override
+        public Operator createOperator(DriverContext driverContext)
+        {
+            return new DynamicFilterCollectOperator(
+                    driverContext.addOperatorContext(operatorId, planNodeId, DynamicFilterCollectOperator.class.getSimpleName()),
+                    dynamicFilter,
+                    types);
+        }
+
+        @Override
+        public void noMoreOperators()
+        {
+        }
+
+        @Override
+        public OperatorFactory duplicate()
+        {
+            throw new UnsupportedOperationException("duplicate() is not supported for DynamicFilterCollectOperatorFactory");
+        }
+    }
+
+    private final OperatorContext context;
+    private boolean finished;
+    private Page current;
+    private final Consumer<TupleDomain<DynamicFilters.Descriptor>> dynamicPredicateConsumer;
```

Comment:
I think it should be `Consumer<TupleDomain<String>>` where `String` is key from `io.prestosql.sql.planner.plan.JoinNode#dynamicFilters`

`DynamicFilters.Descriptor` is specific to probe side

Reply:
Good point, thanks! 
Changed `DynamicFilterCollectOperator` so it won't have to work with `Descriptor`s.